backend/src/post/post.controller.ts

```diff
@@ -1,25 +1,34 @@
-import { Controller, Query, Get, Post, Body, HttpCode, HttpStatus, Delete } from '@nestjs/common';
+import { Controller, Query, Get, Post, Body, HttpCode, HttpStatus, Delete, UseGuards, Request } from '@nestjs/common';
 import { PostService } from './post.service';
 import { createPostDTO } from 'src/dto/post.dto';
+import { AuthGuard } from '../auth/auth.guard';
+import { JwtPayload } from 'src/types/user';
 
 @Controller('posts')
 export class PostController {
     constructor(private readonly postService: PostService) {}
 
     @Post()
     @HttpCode(HttpStatus.CREATED)
-    async createPost(@Body() createPostDTO: createPostDTO){
-      const newPost = await this.postService.createPost(createPostDTO);
+    @UseGuards(AuthGuard) // 認証チェック
+    async createPost(@Body() createPostDTO: createPostDTO, @Request() req: JwtPayload){
+      const user = req.user; // トークンから取得したユーザー情報
+      console.log("check req", req)
+      console.log("test post",user);
+      // print("test post")
+      const newPost = await this.postService.createPost(createPostDTO, user);
       return newPost;
     }
 
     //idがstring型注意,指定の仕方例：/?id=1
     @Delete()
-    deletePost(@Query('id') id:string){
-        return this.postService.deletePost(Number(id));
+    // @UseGuards(AuthGuard) // 認証チェック
```

Comment:
コメントアウト外すと想定通りの挙動をする
```suggestion
    @UseGuards(AuthGuard) // 認証チェック
```